### DISCUSSION TITLE
Validate duplicate path of routes

### DISCUSSION THREAD

**Discussion by iudeen** [encode/starlette#2369] [Ideas] [answered]
I have originally raised a PR in FastAPI for this: https://github.com/tiangolo/fastapi/pull/5595

The PR aims to throw an error when two routes (functions) have same path defined. 

I am starting to feel this would be a nice to have check on Starlette as we head towards 1.0. 

Other frameworks like Flask have such a check as well.

Let me know your thoughts. I am happy to raise a PR if you agree to this.

**Comment by Kludex** [comment, marked as answer]
I don't think it's necessary.

**Reply by Kludex** [reply]
But... If the implementation is simple enough, maybe...
